### PR TITLE
✨ Analog information decoding

### PR DESCRIPTION
**Pull request by lucasberent**
## Description

This PR integrates the newest analog information decoding scripts. 

## Checklist:

- [x] The pull request only contains commits that are related to it.
- [x] I have added appropriate tests and documentation.
- [x] I have made sure that all CI jobs on GitHub pass.
- [x] The pull request introduces no new warnings and follows the project's style guidelines.

### CODE REVIEW

**Comment by lucasberent** [comment]
disabling numba due to https://github.com/vc1492a/PyNomaly/issues/32

File: pyproject.toml [hunk 1]
```diff
@@ -43,11 +44,14 @@ dependencies = [
...
     "qiskit-aer>=0.12.1",
+    "bposd",
+    "numba",
+    "pymatching",
 ]
 dynamic = ["version"]
 
 [project.optional-dependencies]
-test = ["pytest>=7", "pytest-console-scripts>=1.4", "pytest-mock"]
+test = ["pytest>=7", "pytest-console-scripts>=1.4", "pytest-mock", "numba", "pymatching"]
```
Comment (burgholzer) [inline pyproject.toml:54]:
```suggestion
test = ["pytest>=7", "pytest-console-scripts>=1.4", "pytest-mock"]
```

File: pyproject.toml [hunk 2]
```diff
@@ -63,6 +67,8 @@ docs = [
     "sphinxext-opengraph",
     "sphinx-autodoc-typehints",
     "qiskit-terra[visualization]",
+    "numba",
+    "pymatching",
```
Comment (burgholzer) [inline pyproject.toml:71]:
```suggestion
```

File: pyproject.toml [hunk 3]
```diff
@@ -146,7 +152,7 @@ log_cli_level = "INFO"
 xfail_strict = true
 filterwarnings = [
     "error",
-    "ignore:pkg_resources.*:DeprecationWarning:",
+    "ignore:.*pkg_resources.*:DeprecationWarning:pkg_resources",
```
Comment (burgholzer) [inline pyproject.toml:155]:
```suggestion
    "ignore:.*pkg_resources.*:DeprecationWarning:",
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -38,6 +38,8 @@ jobs:
         uses: rui314/setup-mold@v1
       - name: Install Ninja
         run: pipx install ninja
+      - name: Install bposd
+        run: pip install bposd
```
Comment (burgholzer) [inline .github/workflows/ci.yml:42]:
```suggestion
```

**Review by burgholzer — COMMENTED** [review]
Just some quick corrections.

**Review by burgholzer — APPROVED** [review]
Just finished some small cleanups. Brought all external dependencies up to date, enabled formatting for all files and linting for a couple more files than before.

I haven't really checked the code itself, but I have sufficient trust that it is doing what it intends to do. So once CI is happy, I'll go ahead and merge 🎉

**Pull request merged**